provision/kubernetes/job.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2022 tsuru authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/tsuru/tsuru/provision"
+	jobTypes "github.com/tsuru/tsuru/types/job"
+	batchv1 "k8s.io/api/batch/v1"
+	apiv1beta1 "k8s.io/api/batch/v1beta1"
+	apiv1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const jobPrefix = "tsuru-"
+
+func createJobSpec(containerInfo jobTypes.ContainerInfo, labels, annotations map[string]string) batchv1.JobSpec {
+	return batchv1.JobSpec{
+		Template: v1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels:      labels,
+				Annotations: annotations,
+			},
+			Spec: v1.PodSpec{
+				RestartPolicy: "OnFailure",
+				Containers: []v1.Container{
+					{
+						Name:    containerInfo.Name,
+						Image:   containerInfo.Image,
+						Command: containerInfo.Command,
+					},
+				},
+			},
+		},
+	}
+}
+
+func createCronjob(ctx context.Context, client *ClusterClient, job provision.Job, jobSpec batchv1.JobSpec, labels, annotations map[string]string) (string, error) {
+	k8sCronjob, err := client.BatchV1beta1().CronJobs(client.Namespace()).Create(ctx, &apiv1beta1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        job.GetName(),
+			Labels:      labels,
+			Annotations: annotations,
+		},
+		Spec: apiv1beta1.CronJobSpec{
+			Schedule: job.GetSchedule(),
+			JobTemplate: apiv1beta1.JobTemplateSpec{
+				Spec: jobSpec,
+			},
+		},
+	}, metav1.CreateOptions{})
+	return k8sCronjob.Name, err
+}
+
+func createJob(ctx context.Context, client *ClusterClient, job provision.Job, jobSpec batchv1.JobSpec, labels map[string]string, annotations map[string]string) (string, error) {
+	k8sJob, err := client.BatchV1().Jobs(client.Namespace()).Create(ctx, &batchv1.Job{
```

Comment:
Like I said above, you should use `client.PoolNamespace` method to determine the correct namespace.